apps/ui/src/components/sidebar/views/history.tsx

```diff
@@ -1,74 +1,86 @@
 import { UnixNanoTimeStamp } from "@graphql-debugger/time";
 
-import { useContext } from "react";
+import { useContext, useState } from "react";
 import { Link, useParams } from "react-router-dom";
 
+import { SearchBox } from "../../../components/search-box";
 import { ClientContext } from "../../../context/client";
 import { Delete } from "../../../icons/delete";
 import { IDS } from "../../../testing";
+import { traceNameIncludes } from "../../../utils/find-traces";
 import { isTraceError } from "../../../utils/is-trace-error";
 import { rootSpanName } from "../../../utils/root-span-name";
 
 export function History() {
   const params = useParams<{ traceId: string }>();
   const { historyTraces, handleDeleteHistoryTrace } = useContext(ClientContext);
+  const [searchHistory, setSearchHistory] = useState("");
 
   const sortedHistoryTraces = historyTraces.sort((a, b) => {
     return new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime();
   });
 
+  const filteredHistoryTraces = sortedHistoryTraces.filter((historyTrace) =>
+    traceNameIncludes(historyTrace.trace, searchHistory),
+  );
+
   return (
-    <div
-      id={IDS.sidebar.views.history}
-      className="flex w-full flex-col gap-3 divide-y-2 divide-accent"
-    >
-      {sortedHistoryTraces.map(({ schemaId, trace, uniqueId }) => {
-        const durationUnixNano = UnixNanoTimeStamp.fromString(
-          trace.rootSpan?.durationNano || "0",
-        );
+    <>
+      <SearchBox handleSearch={(value) => setSearchHistory(value)} />
+      <div
+        id={IDS.sidebar.views.history}
+        className="flex w-full flex-col gap-3 divide-y-2 divide-accent"
+      >
+        {filteredHistoryTraces.map(({ schemaId, trace, uniqueId }) => {
```

Comment:
What happens if no trace is found?

Reply:
**No traces found** will be displayed. 
![image](https://github.com/rocket-connect/graphql-debugger/assets/45063811/821f400e-ce1b-4f24-b2aa-452a63c2612e)